DQM/L1TMonitor/src/L1TObjectsTiming.cc

```diff
@@ -0,0 +1,480 @@
+    etsum_eta_phi_MHTHF_isolated.at(i)->setAxisTitle("#phi", 1);
+  }
+
+  
+  ibooker.setCurrentFolder(monitorDir+"/L1TMuon"+"/timing"+"/First_bunch");
+  for(unsigned int i=0; i<bxrange-2; ++i) {
+    muons_eta_phi_firstbunch.push_back(ibooker.book2D("muons_eta_phi_bx_firstbunch_"+bx_obj[i],"L1T Muon #eta vs #phi for first bunch BX="+bx_obj[i],25, -2.5, 2.5, 25, -3.2, 3.2));
+    muons_eta_phi_firstbunch.at(i)->setAxisTitle("#eta", 1);
+    muons_eta_phi_firstbunch.at(i)->setAxisTitle("#phi", 2);
+  }
+   
+  ibooker.setCurrentFolder(monitorDir+"/L1TJet"+"/timing"+"/First_bunch");
+  for(unsigned int i=0; i<bxrange-2; ++i) {
+    jet_eta_phi_firstbunch.push_back(ibooker.book2D("jet_eta_phi_bx_firstbunch_"+bx_obj[i],"L1T Jet #eta vs #phi for first bunch BX="+bx_obj[i],25, -2.5, 2.5, 25, -3.2, 3.2));
+    jet_eta_phi_firstbunch.at(i)->setAxisTitle("#eta", 1);
+    jet_eta_phi_firstbunch.at(i)->setAxisTitle("#phi", 2);
+  }
+  
+  ibooker.setCurrentFolder(monitorDir+"/L1TEGamma"+"/timing"+"/First_bunch");
+  for(unsigned int i=0; i<bxrange-2; ++i) {
+    egamma_eta_phi_firstbunch.push_back(ibooker.book2D("egamma_eta_phi_bx_firstbunch_"+bx_obj[i],"L1T EGamma #eta vs #phi for first bunch BX="+bx_obj[i],25, -2.5, 2.5, 25, -3.2, 3.2));
+    egamma_eta_phi_firstbunch.at(i)->setAxisTitle("#eta", 1);
+    egamma_eta_phi_firstbunch.at(i)->setAxisTitle("#phi", 2);
+  }
+
+  ibooker.setCurrentFolder(monitorDir+"/L1TTau"+"/timing"+"/First_bunch");
+  for(unsigned int i=0; i<bxrange-2; ++i) {
+    tau_eta_phi_firstbunch.push_back(ibooker.book2D("tau_eta_phi_bx_firstbunch_"+bx_obj[i],"L1T Tau #eta vs #phi for first bunch BX="+bx_obj[i],25, -2.5, 2.5, 25, -3.2, 3.2));
+    tau_eta_phi_firstbunch.at(i)->setAxisTitle("#eta", 1);
+    tau_eta_phi_firstbunch.at(i)->setAxisTitle("#phi", 2);
+  }
+
+  ibooker.setCurrentFolder(monitorDir+"/L1TEtSum"+"/timing"+"/First_bunch");
+  for(unsigned int i=0; i<bxrange-2; ++i) {
+    etsum_eta_phi_MET_firstbunch.push_back(ibooker.book1D("etsum_phi_bx_MET_firstbunch_"+bx_obj[i],"L1T MET #phi for firstbunch bunch BX="+bx_obj[i],25, -3.2, 3.2));
+    etsum_eta_phi_MET_firstbunch.at(i)->setAxisTitle("#phi", 1);
+    etsum_eta_phi_METHF_firstbunch.push_back(ibooker.book1D("etsum_phi_bx_METHF_firstbunch_"+bx_obj[i],"L1T METHF #phi for firstbunch bunch BX="+bx_obj[i],25, -3.2, 3.2));
+    etsum_eta_phi_METHF_firstbunch.at(i)->setAxisTitle("#phi", 1);
+    etsum_eta_phi_MHT_firstbunch.push_back(ibooker.book1D("etsum_phi_bx_MHT_firstbunch_"+bx_obj[i],"L1T MHT #phi for firstbunch bunch BX="+bx_obj[i],25, -3.2, 3.2));
+    etsum_eta_phi_MHT_firstbunch.at(i)->setAxisTitle("#phi", 1);
+    etsum_eta_phi_MHTHF_firstbunch.push_back(ibooker.book1D("etsum_phi_bx_MHTHF_firstbunch_"+bx_obj[i],"L1T MHTHF #phi for firstbunch bunch BX="+bx_obj[i],25, -3.2, 3.2));
+    etsum_eta_phi_MHTHF_firstbunch.at(i)->setAxisTitle("#phi", 1);
+  }
+
+  
+  ibooker.setCurrentFolder(monitorDir+"/L1TMuon"+"/timing"+"/Last_bunch");
```

Comment:
Only create and fill these MEs if the last bunch trigger bit is was found. So `algoBitLastBxInTrain_ > -1`